addons/udes_stock/models/stock_location.py

```diff
@@ -172,14 +173,46 @@ def set_u_heatmap_data_updated(self, vals):
             bypass_heatmap_check=True
         ).u_heatmap_data_updated = PreciseDatetime.now()
 
+    @api.constrains("usage", "location_id", "child_ids")
+    def _restrict_internal_location_children(self):
+        """
+        Raises a ValidationError if any of the following conditions are met:
+            1. Location is a child of an internal location
+            2. Location is an internal location and has child locations
+        """
+        Location = self.env["stock.location"]
+        for location in self:
+            location = location.with_context(prefetch_fields=False)
```

Comment:
Wouldn't `self = self.with_context(prefetch_fields=False)` achieve the same result with only one context change?

Reply:
Changed.